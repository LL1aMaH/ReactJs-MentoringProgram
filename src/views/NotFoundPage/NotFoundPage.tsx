import React, { useCallback } from 'react';

import { Button } from 'Components/Button';
import { Logo } from 'Components/Logo';

import logo404 from 'Assets/pictures/logo404.jpg';
import { useHistory } from 'react-router-dom';

import styles from './NotFoundPage.css';

const NotFoundPage = (): JSX.Element => {
  const history = useHistory();
  const onClick = useCallback(() => {
    history.push('/');
  }, []);
  return (
    <>
      <div className={styles.notFound}>
        <Logo className={styles.logoHeader} />
        <div className={styles.info}>
          <p className={styles.paragraph}>Page Not Found</p>
          <img src={logo404} alt="404" className={styles.logo404} />
          <Button className={styles.button} onClick={onClick}>
            GO BACK TO HOME
          </Button>
        </div>
      </div>
      <Logo className={styles.logoFooter} />
    </>
  );
};

export default NotFoundPage;
