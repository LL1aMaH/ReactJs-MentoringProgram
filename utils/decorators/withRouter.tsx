import React from 'react';

import { MemoryRouter } from 'react-router-dom';

export type TComponentFunc = () => JSX.Element;
export type TDecoratorLocation = {
  pathname?: string;
  search?: string;
};

const withRouter = (
  componentFunc: TComponentFunc,
  location: TDecoratorLocation = { pathname: '/' },
): JSX.Element => {
  return <MemoryRouter initialEntries={[location]}>{componentFunc()}</MemoryRouter>;
};

export default withRouter;
