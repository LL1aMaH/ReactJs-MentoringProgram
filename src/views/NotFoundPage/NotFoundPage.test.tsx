import React from 'react';
import { render } from '@testing-library/react';

import withRouter from '../../../utils/decorators/withRouter';

import NotFoundPage from './NotFoundPage';

describe('NotFoundPage', () => {
  it('renders correctly', () => {
    const component = render(withRouter(() => <NotFoundPage />));
    expect(component.asFragment()).toMatchSnapshot();
  });
});
