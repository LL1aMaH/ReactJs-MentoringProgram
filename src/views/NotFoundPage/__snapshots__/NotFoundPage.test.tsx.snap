// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="notFound"
  >
    <a
      class="logo logoHeader"
      href="/"
    >
      <b>
        netflix
      </b>
      roulette
    </a>
    <div
      class="info"
    >
      <p
        class="paragraph"
      >
        Page Not Found
      </p>
      <img
        alt="404"
        class="logo404"
        src="[object Object]"
      />
      <button
        class="button button"
        tabindex="0"
      >
        GO BACK TO HOME
      </button>
    </div>
  </div>
  <a
    class="logo logoFooter"
    href="/"
  >
    <b>
      netflix
    </b>
    roulette
  </a>
</DocumentFragment>
`;
